src/Grid/LocalDataSource.ts

```diff
@@ -0,0 +1,311 @@
+import Axios from 'axios';
+import * as _ from 'lodash';
+import * as moment from 'moment';
+import * as Rx from 'rx';
+import { ColumnModel } from '.';
+import {
+  AggregateFunctions,
+  ColumnDataType,
+  ColumnSortDirection,
+  CompareOperators,
+} from './Column';
+import GridRequest from './GridRequest';
+import GridResponse from './GridResponse';
+import GridDataResponse from './utils/GridDataResponse';
+
+export default class LocalDataSource implements IDataSource {
+
+  public static defaultColumnValues = {
+    Aggregate: AggregateFunctions.NONE,
+    DataType: ColumnDataType.STRING,
+    IsKey: false,
+    Searchable: false,
+    Sortable: false,
+    Visible: true
+  };
+
+  public columns: ColumnModel[];
+  public dataStream: any;
+  public localData: any[];
+  public counter: number;
+
+  constructor(localData: any[], columns: ColumnModel[]) {
+    this.localData = localData;
+    this.dataStream = new Rx.BehaviorSubject({ Payload: [] });
+    this.columns = columns;
+    this.counter = 0;
+  }
+
+  public connect(rowsPerPage: number, page: number, searchText: string) {
+    this._updateDataStream(rowsPerPage, page, searchText);
+    return this.dataStream;
+  }
+
+  public refresh(rowsPerPage: number, page: number, searchText: string) {
+    this._updateDataStream(rowsPerPage, page, searchText);
+  }
+
+  public getAllRecords = (rowsPerPage: number, page: number, searchText: string): Promise<object> =>
+    new Promise((resolve, reject) => {
+      let data = this.localData;
+
+      const request = new GridRequest({
+        Columns: this.columns,
+        Count: this.counter++,
```

Comment:
The Count should be an internal static variable and when the constructor is called increase one and use it.